packages/core/src/utils/userop.ts

```diff
@@ -141,11 +142,13 @@ export function applyUserOpFeeOption(
  * @param [feeOption] the fee option field value to apply
  * @returns the new value of the field after applying the override or fee option
  */
-export function applyUserOpOverrideOrFeeOption(
-  value: BigNumberish | undefined,
+export function applyUserOpOverrideOrFeeOption<
+  TValue extends BigNumberish | undefined
+>(
+  value: TValue,
   override?: BigNumberish | Multiplier,
   feeOption?: UserOperationFeeOptionsField
-): BigNumberish {
+): TValue | BigNumberish {
```

Comment:
Any reason to make this not just return `TValue`? I could be missing something obvious lol

Reply:
typescript doesn't like that BigNumberish and TValue overlap but TValue can be instantiated with something other than BigNumberish